File: Algorithms/Java/Rotate Image.java
public class Solution {
    public void rotate(int[][] matrix) {
		int length = matrix.length;
		for (int i = 0; i < length; i++) {
			for (int j = 0; j < length - i; j++)
				swap(matrix, i, j, length - j - 1, length - i - 1);
		}
		for (int i = 0; i < length / 2; i++)
			for (int j = 0; j < length; j++)
				swap(matrix, i, j, length - i - 1, j);
	}

	public void swap(int[][] matrix, int ai, int aj, int bi, int bj) {
		int temp = matrix[ai][aj];
		matrix[ai][aj] = matrix[bi][bj];
		matrix[bi][bj] = temp;
	}
}

File: Algorithms/Python/Rotate-Image.py
class Solution(object):

    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        if not matrix:
            return
        l = len(matrix)
        for i in range(l):
            for j in range(l - i):
                matrix[i][j], matrix[l - 1 - j][l - 1 -
                                                i] = matrix[l - 1 - j][l - 1 - i], matrix[i][j]
        for i in range(l // 2):
            for j in range(l):
                matrix[i][j], matrix[
                    l - 1 - i][j] = matrix[l - 1 - i][j], matrix[i][j]
